### PR TITLE
[stdlib] Fix _writeBackMutableSlice error messages

### PR DESCRIPTION
**Pull request by natecook1000**
#### What's in this pull request?

The error messages in the `_writeBackMutableSlice` function are reversed. For example:

``` swift
var a = Array(1...20)
a.withUnsafeMutableBufferPointer { buffer in
    buffer[4...15] = buffer[2...3]
}
// fatal error: Cannot replace a slice of a MutableCollection with a slice of a larger size
```
#### Resolved bug number: n/a

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| All supported platforms | @swift-ci Please smoke test and merge |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

**Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| All supported platforms | @swift-ci Please test and merge |
| OS X platform | @swift-ci Please test OS X platform |
| OS X platform | @swift-ci Please benchmark |
| Linux platform | @swift-ci Please test Linux platform |

**Lint Testing**

| Language | Comment |
| --- | --- |
| Python | @swift-ci Please Python lint |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

### CODE REVIEW

**Comment by gribozavr** [comment]
Thank you!  Would you mind writing a test?  It should go into `validation-test/stdlib/CollectionType.swift.gyb`.  You can use `.crashOutputMatches("...")` to check the error message.

**Comment by natecook1000** [comment]
@gribozavr I tried to use `MinimalMutableRandomAccessCollection`, but it wasn't giving me any error output to check, so I'm using the code from the example above. Does that work?

File: validation-test/stdlib/CollectionType.swift.gyb [hunk 1]
```diff
@@ -588,6 +588,26 @@ CollectionTypeTests.test("subscript(_: Range<Index>)/writeback") {
     [ 1, 2, 3, 4, 5, 0, -1, -2, -3, -4 ], collection)
 }
 
+CollectionTypeTests.test("subscript(_: Range<Index>)/sliceTooLarge")
```
Comment (gribozavr) [inline validation-test/stdlib/CollectionType.swift.gyb]: Could you clarify that you are testing the default implementation?

```
CollectionTypeTests.test("subscript(_: Range<Index>)/defaultImplementation/sliceTooLarge")
```

File: validation-test/stdlib/CollectionType.swift.gyb [hunk 2]
```diff
@@ -588,6 +588,26 @@ CollectionTypeTests.test("subscript(_: Range<Index>)/writeback") {
...
+  .code {
+  var x = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+  expectCrashLater()
+  x.withUnsafeMutableBufferPointer { buffer in
+    buffer[2..<4] = buffer[4..<8]
+  }
+}
+
+CollectionTypeTests.test("subscript(_: Range<Index>)/sliceTooSmall")
+  .crashOutputMatches("Cannot replace a slice of a MutableCollection with a slice of a smaller size")
```
Comment (gribozavr) [inline validation-test/stdlib/CollectionType.swift.gyb]: And here.

**Comment by gribozavr** [comment]
> I tried to use `MinimalMutableRandomAccessCollection`, but it wasn't giving me any error output to check, so I'm using the code from the example above. Does that work?

The reason it does not work is that StdlibCollectionUnittest is built in optimized mode, and because of that the error messages get eliminated.

**Comment by gribozavr** [comment]
@swift-ci Please test

**Comment by gribozavr** [comment]
The change LGTM, only one nitpicky comment :)

**Comment by natecook1000** [comment]
@gribozavr Fixed!

**Comment by gribozavr** [comment]
@swift-ci Please test and merge

**Pull request merged**